src/Sources/AbstractSource.php

```diff
@@ -135,24 +135,25 @@ function ($migration_class) {
         $this->onUnInstall();
     }
 
-    protected function save(Model $me, Model $primary)
+    protected function save(Model $source, $primary = null)
     {
-        // @todo: update $me name
         // @todo: Only save model if something has changed
         // @todo: in the Audit, describe what changed
+
         // Now, we save the primary version
-        $primary->save();
+        if($primary !== null){
+            $primary->save();
+            $source->game_id = $primary->id;
+        }
 
         //@todo: update game_id to more generic
-        $me->game_id = $primary->id;
-
         // Create new record in this source
-        $me->save();
+        $source->save();
 
         // Create a new Audit Record because something changed
         $audit = new Audit();
         $audit->job_id = $this->getJob()->id;
-        $audit->record_id = $me->id;
+        $audit->record_id = $source->id;
         $audit->source_key = static::getKey();
```

Comment:
Here is where we will (eventually) want to add more.

```
foreach ($what_changed as $change) {
       $audit = new Audit();
        $audit->job_id = $this->getJob()->id;
        $audit->record_id = $source->id;
        $audit->source_key = static::getKey();
        $audit->column = $change['column'];
        $audit->previous = $change['previous'];
        $audit->new = $change['new'];
}
```

OR

```php
       $audit = new Audit();
        $audit->job_id = $this->getJob()->id;
        $audit->record_id = $source->id;
        $audit->source_key = static::getKey();
        $audit->changes = json_encode($what_changed); 
```

In both cases `$what_changed` will be an array with the column that changed, the old value, and the new value